File: ios_app/iphone/NoteSheet.swift
import SwiftUI
import shared

struct NoteSheet: View {

    @EnvironmentObject private var nativeSheet: NativeSheet

    @Binding private var isPresented: Bool
    @State private var vm: NoteSheetVm

    init(
            isPresented: Binding<Bool>,
            initNote: NoteDb
    ) {
        _isPresented = isPresented
        _vm = State(initialValue: NoteSheetVm(note: initNote))
    }

    var body: some View {

        VMView(vm: vm, stack: .VStack()) { state in

            ScrollView {

                VStack {

                    Padding(vertical: 16)

                    Text(state.note.text)

                    Padding(vertical: 16)
                }
                        .padding(.horizontal, H_PADDING)
            }

            Spacer()

            Sheet__BottomView {

                HStack {

                    Spacer()

                    Sheet__BottomView__SecondaryButton(text: "Edit") {
                        nativeSheet.show { isEditPresented in
                            NoteFormSheet(
                                    isPresented: isEditPresented,
                                    note: state.note,
                                    onDelete: { isPresented = false }
                            )
                        }
                    }

                    Sheet__BottomView__SecondaryButton(text :"Close") {
                        isPresented = false
                    }
                }
                        .padding(.top, 10)
                        .padding(.trailing, H_PADDING - 8)
                        .padding(.bottom, 10)
            }
        }
    }
}

File: android_app/src/main/java/me/timeto/app/ui/NoteSheet.kt
package me.timeto.app.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import me.timeto.app.*
import me.timeto.shared.db.NoteDb
import me.timeto.shared.vm.NoteSheetVM

@Composable
fun NoteSheet(
    layer: WrapperView.Layer,
    initNote: NoteDb, // TRICK Can change, use VM!
) {

    val (_, state) = rememberVM(initNote) { NoteSheetVM(initNote) }

    VStack(
        modifier = Modifier
            .fillMaxHeight()
            .background(c.sheetBg)
    ) {

        LazyColumn(
            modifier = Modifier
                .fillMaxWidth()
                .weight(1f),
            contentPadding = PaddingValues(
                top = 12.dp,
                bottom = 12.dp,
            ),
        ) {
            item {
                Text(
                    text = state.note.text,
                    modifier = Modifier
                        .padding(horizontal = H_PADDING),
                    color = c.text,
                )
            }
        }

        Sheet__BottomView {
            HStack(
                modifier = Modifier
                    .padding(
                        top = 10.dp,
                        end = H_PADDING - 8.dp,
                        bottom = 10.dp,
                    ),
                verticalAlignment = Alignment.CenterVertically,
            ) {

                SpacerW1()

                Sheet__BottomView__SecondaryButton("Edit") {
                    Sheet.show { layerForm ->
                        NoteFormSheet(
                            layer = layerForm,
                            note = state.note,
                            onDelete = { layer.close() },
                        )
                    }
                }

                Sheet__BottomView__SecondaryButton("Close") {
                    layer.close()
                }
            }
        }
    }
}
